### PR TITLE
Leopoly_Phase1_001_QMLWrapper

### PR DESCRIPTION
**Pull request by humbletim**
... PR staging; description and test plan pending

To fix QML threading issues that affect the whole app

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5773/app_12034_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5773/HighFidelity-Beta-PR12034-be11157cd1ba2455ee13fb2fbfbdcb9186c527f6.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5773/HighFidelity-Beta-PR12034-be11157cd1ba2455ee13fb2fbfbdcb9186c527f6.exe

**Comment by MiladNazeri** [comment]
Overall Leopoly PR roadmap:
https://docs.google.com/document/d/1GJC6qYHICYccIqdmaGgtt30Nj12Q3FtFmejns7biyks/edit
This is 1/9

The change is 4 lines of code so not sure if an extensive test plan is needed here.  
Labeling Do not merge until code is reviewed for #'s 1-8

**Comment by ZappoMan** [comment]
How do we develop a test plan for this that verifies the threading changes won't cause additional dead locks?

File: libraries/ui/src/ui/QmlWrapper.cpp [hunk 1]
```diff
@@ -20,14 +20,16 @@ QmlWrapper::QmlWrapper(QObject* qmlObject, QObject* parent)
 
 void QmlWrapper::writeProperty(QString propertyName, QVariant propertyValue) {
     if (QThread::currentThread() != thread()) {
-        QMetaObject::invokeMethod(this, "writeProperty", Q_ARG(QString, propertyName), Q_ARG(QVariant, propertyValue));
+        BLOCKING_INVOKE_METHOD(this, "writeProperty", Q_ARG(QString, propertyName), Q_ARG(QVariant, propertyValue));
```
Comment (humbletim) [inline libraries/ui/src/ui/QmlWrapper.cpp]: The blocking invoke is needed to keep QML Properties consistent (eg: writing a value and then immediately reading it back), but fixing the crashing problem itself only relies on the missing `return;` statements.

Would it be better to limit the PR to just that `return;` statement for now?

**Comment by humbletim** [comment]
updated PR to only include the two missing `return` statements which are needed to avoid crashing

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6035/app_12034_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6035/HighFidelity-Beta-PR12034-175db84d3bcbd1b873533b201a305def69ca4055.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6035/HighFidelity-Beta-PR12034-175db84d3bcbd1b873533b201a305def69ca4055.dmg

**Comment by MiladNazeri** [comment]
closing this PR as this most likely isn't needed anymore.

**Pull request closed without merging**